File: src/main/java/com/freya02/bot/tag/ShortTag.java
package com.freya02.bot.tag;

import java.lang.reflect.RecordComponent;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Arrays;

public record ShortTag(String name, String description) {
	public static final String[] COLUMN_NAMES = Arrays.stream(ShortTag.class.getRecordComponents())
			.map(RecordComponent::getName)
			.toArray(String[]::new);

	public static ShortTag fromResult(ResultSet set) throws SQLException {
		return new ShortTag(
				set.getString("name"),
				set.getString("description")
		);
	}
}

File: doxxy-bot/src/main/kotlin/dev/freya02/doxxy/bot/tag/ShortTag.kt
package dev.freya02.doxxy.bot.tag

import java.sql.ResultSet
import java.sql.SQLException

class ShortTag(val name: String, val description: String) {
    companion object {

        @Throws(SQLException::class)
        fun fromResult(set: ResultSet): ShortTag {
            return ShortTag(
                set.getString("name"),
                set.getString("description")
            )
        }
    }
}
